### ISSUE TITLE
0 counts for k-mer generated from same sequence as used in FM-index

### ISSUE THREAD

**Issue by EricR86** [TravisWheelerLab/AvxWindowFmIndex#41]
Hello,

I came across an odd bug. I created an index for mm39. I then generated a k-mer sequence from [mm39](https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_000001635.27/) chr19:
'GGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGG'

While very repetitive, the `awFmParallelSearchCount` function returns 0 for this k-mer in the search list.

I can confirm the sequence exists inside chr19:

From lines 766498 onward (refseq: NC_000085.7):
```
                                                                        *
TTCACAGGAATACCCCACTCTGCTGGTACCAATTTGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGG
TTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTT
AGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAG
GGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGG
```
I marked the beginning of the sub-sequence with a `*`

I am on the latest commit. I have not regenerated the index since commit [71e7dd6](https://github.com/TravisWheelerLab/AvxWindowFmIndex/tree/71e7dd684f935f43ee0de28cfc519bbe7422d808). There were 5 of these total I found in the entire genome, all very long k-mers of length 250-ish or over. This was just the example I managed to track down

**Comment by EricR86** [comment]
It is worth noting that adding a letter to the problematic k-mer sequence results in a count of 1, and removing a letter from the problematic k-mer sequence results in a count of 0.

**Comment by Sawwave** [comment]
Thanks for the bug report. I'm currently wrapping up my  Ph.D dissertation so my time is limited, but I'll address this as soon as I have the time.

**Comment by Sawwave** [comment]
Now that I have time to look at this, I've confirmed that I can reproduce the error. I'm working to identify the problem, and I hope to have this fixed soon.

**Comment by Sawwave** [comment]
I've solved this issue merge with #42 . There was an issue with some query position variables using uint8_t's, and having issues with queries longer than 255. These variables have been updated to uint32_t's, so they should handle even unreasonably long queries. 

Thanks again for reporting this error!
